presto-main/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -420,4 +440,29 @@ public static RowExpression coalesce(List<RowExpression> expressions)
     {
         return new SpecialFormExpression(SpecialFormExpression.Form.COALESCE, expressions.get(0).getType(), expressions);
     }
+
+    public static boolean isSupportedArrayContainsFilter(FunctionResolution functionResolution, RowExpression filterExpression, List<VariableReferenceExpression> left, List<VariableReferenceExpression> right)
+    {
+        List<Type> supportedTypes = ImmutableList.of(BIGINT, INTEGER, VARCHAR, DATE);
+
+        if (filterExpression instanceof CallExpression) {
+            CallExpression callExpression = (CallExpression) filterExpression;
+            if (functionResolution.isArrayContainsFunction(callExpression.getFunctionHandle())) {
+                RowExpression array = callExpression.getArguments().get(0);
+                RowExpression element = callExpression.getArguments().get(1);
```

Comment:
How about also add a check here: `if (!(element instance of VariableReferenceExpression)) return false;` so that we filter out this case earlier.

Reply:
I changed the check below to ```right.contains(element)``` which would fail if element is not a VariableReferenceExpression